src/hashtree.erl

```diff
@@ -966,13 +994,19 @@ iterate({ok, K, V}, IS=#itr_state{itr=Itr,
                                segment_acc=[{K,V}],
                                prefetch=true},
             iterate(iterator_move(Itr, prefetch), IS2);
-        {Id, _, [_NextSeg | _Remaining], true} ->
+        {Id, _, [NextSeg | Remaining], true} ->
             %% Pointing at uninteresting segment, but need to halt the
             %% prefetch to ensure the interator can be reused
-            IS2 = IS#itr_state{segment_acc=[],
+
+            %% Do not update segment/final_acc
+            IS2 = IS#itr_state{current_segment=NextSeg,
+                               segment_acc=[],
+                               remaining_segments=Remaining,
                                final_acc=[{Segment, F(Acc)} | FinalAcc],
-                               prefetch=false},
-            iterate(iterator_move(Itr, prefetch_stop), IS2);
+                               prefetch=true}, % will be after second move
+            _ = iterator_move(Itr, prefetch_stop), % ignore the pre-fetch,
```

Comment:
I think I understand everything in this PR except for this one bit here. In the old code, we recurse on the result of prefetch_stop in case it happens to drop us off at the beginning of the next interesting segment. Why can't we continue to do that? In the common case where we don't get to an interesting segment, I would think we'd just fall through to the prefetch=false clause below and then seek to the next segment from there.

The comment acknowledges we're risking a reseek, but doesn't explain why we need to take that risk. Maybe it's just not that big a risk though, and it's a cheap operation regardless, so the code is just simpler this way? But either way, I'd like to know what motivated the change.